web/src/lib/components/shared-components/notification/notification-card.svelte

```diff
@@ -51,26 +52,43 @@
 	let removeNotificationTimeout: NodeJS.Timeout | undefined = undefined;
 
 	onMount(() => {
-		removeNotificationTimeout = setTimeout(() => {
-			notificationController.removeNotificationById(notificationInfo.id);
-		}, notificationInfo.timeout);
-
+		removeNotificationTimeout = setTimeout(discard, notificationInfo.timeout);
 		return () => clearTimeout(removeNotificationTimeout);
 	});
+
+	const discard = () => {
+		notificationController.removeNotificationById(notificationInfo.id);
+	};
+
+	const handleClick = (event: MouseEvent) => {
+		const action = notificationInfo.action;
+		if (action.type == 'discard') {
+			discard();
+		} else if (action.type == 'link') {
+			window.open(action.target);
+			event.preventDefault();
+		}
+	};
 </script>
 
 <div
 	transition:fade={{ duration: 250 }}
 	style:background-color={backgroundColor()}
 	style:border={borderStyle()}
-	class="min-h-[80px] w-[300px] rounded-2xl z-[999999] shadow-md p-4 mb-4"
+	class="min-h-[80px] w-[300px] rounded-2xl z-[999999] shadow-md p-4 mb-4 hover:cursor-pointer"
+	on:click={handleClick}
 >
-	<div class="flex gap-2 place-items-center">
-		<svelte:component this={icon} color={primaryColor()} size="20" />
-		<h2 style:color={primaryColor()} class="font-medium" data-testid="title">
-			{notificationInfo.type.toString()}
-		</h2>
+	<div class="flex justify-between">
+		<div class="flex gap-2 place-items-center">
+			<svelte:component this={icon} color={primaryColor()} size="20" />
+			<h2 style:color={primaryColor()} class="font-medium" data-testid="title">
+				{notificationInfo.type.toString()}
+			</h2>
+		</div>
+		<button on:click={discard}>
+			<svelte:component this={WindowClose} size="20" />
+		</button>
 	</div>
 
-	<p class="text-sm pl-[28px] pr-[16px]" data-testid="message">{notificationInfo.message}</p>
+	<p class="text-sm pl-[28px] pr-[16px]" data-testid="message">{@html notificationInfo.message}</p>
```

Comment:
with the `@html` tag, it should render the string with valid html syntax, right?

Reply:
It properly renders html tags, yes.